pkg/odo/cli/component/push.go

```diff
@@ -65,6 +66,13 @@ func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string)
 		}
 		po.EnvSpecificInfo = envinfo
 		po.Context = genericclioptions.NewDevfileContext(cmd)
+
+		if cmd.Flags().Changed("project") {
```

Comment:
I think this needs to be moved here https://github.com/openshift/odo/blob/6740e5177ae11db734f7d3d303c4f689015e3b91/pkg/odo/genericclioptions/context.go#L343. We are repeating the same steps in all the above files.

Reply:
@mik-dass You're right, good catch!

Reply:
Actually, we don't need to move any code to `resolveNamespace`, just update it to retrieve the namespace from the `--project` flag.

So long as we initialize the devfile context in each command that needs it, we can retrieve the namespace via `kclient.Namespace`